src/shared/parameter_v2/config/examples/parameter.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+// Parameter for C
+//
+// Parameters for C will ALWAYS be immutable. We won't have a UI updating these
+// values on the robot.
+//
+// In the future if we chose to support sending dynamic parameters over the
+// network, we can simply replace the existing param tree on the robot
+// with the updated one.
+//
+// We don't use an psuedo-class here to keep things simple as we can make use of
+// compile time checks to provide immutability. (without hiding the contents of
+// the struct in the c file)
+//
+// It is garunteed that the values don't change after `app_dynamic_parameters_create()`
+// has been called
+//
+// The compiler will NOT allow assignment to these structs without a struct initializer.
+// These struct initializers are auto generated from the yaml, OR can be created manually
+// when needed for testing.
+typedef struct BoolParameter
+{
+    const bool value;
+} BoolParameter_t;
```

Comment:
What was the rationale behind exposing the members of these structs directly, rather then hiding them behind getter functions like we normally do?

Reply:
I wanted `example_config` to have access to the internals to make code generation easier, but after a good nights rest, realized that it doesn't matter LOL, its generated anyway, nobody is writing that manually

Changed to pseudo-class